### PR TITLE
executor, optimizer: add simple plan and executor, implement UseStmt.

### PR DESCRIPTION
**Pull request by coocood**
With `SImple` plan and `SimpleExec` executor, implementing simple statements would be much easier.

### CODE REVIEW

File: executor/executor_simple.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/db"
+	"github.com/pingcap/tidb/sessionctx/variable"
+)
+
+// SimpleExec represents simple statement executor.
+type SimpleExec struct {
```
Comment (shenli) [inline executor/executor_simple.go]: Add some comments

File: executor/executor_simple.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/sessionctx/db"
+	"github.com/pingcap/tidb/sessionctx/variable"
+)
+
+// SimpleExec represents simple statement executor.
+// For statements do simple execution, like `USE`, 'SET` ...
```
Comment (ngaut) [inline executor/executor_simple.go]: What's ... for ? "ping" ? Please add more details.

Comment (coocood) [inline executor/executor_simple.go]: Right now, what statements is simple is not fully decided.
I'll update it after all the statements implemented in new plan.

Comment (ngaut) [inline executor/executor_simple.go]: Please add TODO.

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**